client/damage/damage.lua

```diff
@@ -2,8 +2,15 @@ local config = require 'config.client'
 local sharedConfig = require 'config.shared'
 local playerArmor = nil
 local damageEffectsEnabled = true
-local WEAPONS = exports.qbx_core:GetWeapons()
+local WEAPONS = {}
 
+-- Initialize weapons after ensuring core is loaded
+CreateThread(function()
+    while not exports.qbx_core do
+        Wait(100)
+    end
+    WEAPONS = exports.qbx_core:GetWeapons()
+end)
```

Comment:
This should not be needed as all of these resources require the core as a dependency and are thus loaded after the core.

Reply:
I was facing the issue so resolved it by doing this. And yes it's required. If it not then I should not see the error! 

> I was getting this error : 
> [ Error : [ script:qbx_medical] SCRIPT ERROR: @qbx_medical/client/damage/damage.lua:211: attempt to index a nil value (field '?') [ script:qbx_medical] > checkForDamage (@qbx_medical/client/damage/damage.lua:211) [ script:qbx_medical] > fn (@qbx_medical/client/damage/damage.lua:225)

When I was double jump and fall down on the ground the error comming.

I tested the code and working perfectly and don't get the error while doing the same task.